dev-util/rizin-0.8.1 and dev-util/cutter-2.4.1 version bump

A result of a year of work, Rizin 0.8.1:
- https://github.com/rizinorg/rizin/releases/tag/v0.8.0
- https://github.com/rizinorg/rizin/releases/tag/v0.8.1

And corresponding Cutter release:
- https://github.com/rizinorg/cutter/releases/tag/v2.4.0
- https://github.com/rizinorg/cutter/releases/tag/v2.4.1

Unfortunatly getting a linker error at the end of the build for rz_core :/

```
[1654/1667] cc  -o librz/core/librz_core.so.0.8.1 librz/core/librz_core.so.0.8.1.p/meson-generated_.._cmd_descs_cmd_descs.c.o librz/core/librz_core.so.0.8.1.p/agraph.c.o librz/core/librz_core.so.0.8.1.p/analysis_objc.c.o librz/core/librz_core.so.0.8.1.p/analysis_tp.c.o librz/core/librz_core.so.0.8.1.p/basefind.c.o librz/core/librz_core.so.0.8.1.p/cagraph.c.o librz/core/librz_core.so.0.8.1.p/cgraph.c.o librz/core/librz_core.so.0.8.1.p/canalysis.c.o librz/core/librz_core.so.0.8.1.p/cannotated_code.c.o librz/core/librz_core.so.0.8.1.p/carg.c.o librz/core/librz_core.so.0.8.1.p/casm.c.o librz/core/librz_core.so.0.8.1.p/cautocmpl.c.o librz/core/librz_core.so.0.8.1.p/cbin.c.o librz/core/librz_core.so.0.8.1.p/cbounds.c.o librz/core/librz_core.so.0.8.1.p/cconfig.c.o librz/core/librz_core.so.0.8.1.p/ccrypto.c.o librz/core/librz_core.so.0.8.1.p/cdebug.c.o librz/core/librz_core.so.0.8.1.p/cesil.c.o librz/core/librz_core.so.0.8.1.p/cfile.c.o librz/core/librz_core.so.0.8.1.p/cflag.c.o librz/core/librz_core.so.0.8.1.p/chash.c.o librz/core/librz_core.so.0.8.1.p/cheap.c.o librz/core/librz_core.so.0.8.1.p/cil.c.o librz/core/librz_core.so.0.8.1.p/cio.c.o librz/core/librz_core.so.0.8.1.p/clang.c.o librz/core/librz_core.so.0.8.1.p/cmeta.c.o librz/core/librz_core.so.0.8.1.p/cmp.c.o librz/core/librz_core.so.0.8.1.p/core.c.o librz/core/librz_core.so.0.8.1.p/cparser.c.o librz/core/librz_core.so.0.8.1.p/cpdb.c.o librz/core/librz_core.so.0.8.1.p/cplugin.c.o librz/core/librz_core.so.0.8.1.p/cprint.c.o librz/core/librz_core.so.0.8.1.p/creg.c.o librz/core/librz_core.so.0.8.1.p/csearch.c.o librz/core/librz_core.so.0.8.1.p/csign.c.o librz/core/librz_core.so.0.8.1.p/ctypes.c.o librz/core/librz_core.so.0.8.1.p/cvfile.c.o librz/core/librz_core.so.0.8.1.p/csyscall.c.o librz/core/librz_core.so.0.8.1.p/disasm.c.o librz/core/librz_core.so.0.8.1.p/fortune.c.o librz/core/librz_core.so.0.8.1.p/golang.c.o librz/core/librz_core.so.0.8.1.p/hack.c.o librz/core/librz_core.so.0.8.1.p/libs.c.o librz/core/librz_core.so.0.8.1.p/linux_heap_glibc.c.o librz/core/librz_core.so.0.8.1.p/linux_heap_glibc64.c.o librz/core/librz_core.so.0.8.1.p/rop.c.o librz/core/librz_core.so.0.8.1.p/project.c.o librz/core/librz_core.so.0.8.1.p/project_migrate.c.o librz/core/librz_core.so.0.8.1.p/rtr.c.o librz/core/librz_core.so.0.8.1.p/seek.c.o librz/core/librz_core.so.0.8.1.p/serialize_core.c.o librz/core/librz_core.so.0.8.1.p/task.c.o librz/core/librz_core.so.0.8.1.p/yank.c.o librz/core/librz_core.so.0.8.1.p/p_core_dex.c.o librz/core/librz_core.so.0.8.1.p/p_core_java.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_alias.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_api.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_block.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_cmp.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_egg.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_eval.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_flag.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_flirt.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_hash.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_history.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_info.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interactive.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interactive_panel.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interpret.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_macro.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_meta.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_open.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_plugins.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_project.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_query.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_quit.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_remote.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_resize.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_seek.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_shell.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_system.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_tasks.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_type.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_write.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_yank.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_regs.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_linux_heap_glibc.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_windows_heap.c.o librz/core/librz_core.so.0.8.1.p/tui_biteditor.c.o librz/core/librz_core.so.0.8.1.p/tui_classes.c.o librz/core/librz_core.so.0.8.1.p/tui_colors.c.o librz/core/librz_core.so.0.8.1.p/tui_comments.c.o librz/core/librz_core.so.0.8.1.p/tui_config.c.o librz/core/librz_core.so.0.8.1.p/tui_define.c.o librz/core/librz_core.so.0.8.1.p/tui_esil.c.o librz/core/librz_core.so.0.8.1.p/tui_flags.c.o librz/core/librz_core.so.0.8.1.p/tui_hud.c.o librz/core/librz_core.so.0.8.1.p/tui_modes.c.o librz/core/librz_core.so.0.8.1.p/tui_visual.c.o librz/core/librz_core.so.0.8.1.p/tui_vmenus.c.o librz/core/librz_core.so.0.8.1.p/tui_vmenus_graph.c.o librz/core/librz_core.so.0.8.1.p/tui_vasm.c.o librz/core/librz_core.so.0.8.1.p/tui_vmarks.c.o librz/core/librz_core.so.0.8.1.p/tui_panels.c.o librz/core/librz_core.so.0.8.1.p/tui_tabs.c.o librz/core/librz_core.so.0.8.1.p/tui_rop.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,-soname,librz_core.so.0.8 '-Wl,-rpath,$ORIGIN/../util:$ORIGIN/../demangler:$ORIGIN/../diff:$ORIGIN/../magic:$ORIGIN/../socket:$ORIGIN/../flag:$ORIGIN/../cons:$ORIGIN/../lang:$ORIGIN/../hash:$ORIGIN/../crypto:$ORIGIN/../io:$ORIGIN/../reg:$ORIGIN/../bp:$ORIGIN/../syscall:$ORIGIN/../egg:$ORIGIN/../arch:$ORIGIN/../search:$ORIGIN/../config:$ORIGIN/../bin:$ORIGIN/../type:$ORIGIN/../il:$ORIGIN/../sign:$ORIGIN/../debug:/packages/tree_sitter-0.20.10~git-1/.self/develop/lib:/packages/libpcre2-10.45-1/.self/develop/lib:/packages/zlib-1.3.1-4/.self/develop/lib:/packages/xz_utils-5.6.2-2/.self/develop/lib:/packages/openssl3-3.5.0-2/.self/develop/lib:/packages/file-5.43-2/.self/develop/lib:/packages/xxhash-0.8.3-1/.self/develop/lib:/packages/libzip-1.9.2-2/.self/develop/lib:/packages/capstone5-5.0.1-2/.self/develop/lib:/packages/zstd-1.5.6-2/.self/develop/lib:/packages/libmspack-0.10.1~alpha-2/.self/develop/lib' -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/util -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/demangler -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/diff -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/magic -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/socket -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/flag -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/cons -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/lang -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/hash -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/crypto -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/io -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/reg -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/bp -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/syscall -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/egg -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/arch -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/search -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/config -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/bin -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/type -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/il -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/sign -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/debug '-Wl,-rpath-link,/packages/tree_sitter-0.20.10~git-1/.self/develop/lib' -Wl,-rpath-link,/packages/libpcre2-10.45-1/.self/develop/lib -Wl,-rpath-link,/packages/zlib-1.3.1-4/.self/develop/lib -Wl,-rpath-link,/packages/xz_utils-5.6.2-2/.self/develop/lib -Wl,-rpath-link,/packages/openssl3-3.5.0-2/.self/develop/lib -Wl,-rpath-link,/packages/file-5.43-2/.self/develop/lib -Wl,-rpath-link,/packages/xxhash-0.8.3-1/.self/develop/lib -Wl,-rpath-link,/packages/libzip-1.9.2-2/.self/develop/lib -Wl,-rpath-link,/packages/capstone5-5.0.1-2/.self/develop/lib -Wl,-rpath-link,/packages/zstd-1.5.6-2/.self/develop/lib '-Wl,-rpath-link,/packages/libmspack-0.10.1~alpha-2/.self/develop/lib' -Wl,--start-group librz/util/librz_util.so.0.8.1 librz/demangler/librz_demangler.so.0.8.1 librz/diff/librz_diff.so.0.8.1 librz/magic/librz_magic.so.0.8.1 librz/socket/librz_socket.so.0.8.1 librz/flag/librz_flag.so.0.8.1 librz/cons/librz_cons.so.0.8.1 librz/lang/librz_lang.so.0.8.1 librz/hash/librz_hash.so.0.8.1 librz/crypto/librz_crypto.so.0.8.1 librz/io/librz_io.so.0.8.1 librz/reg/librz_reg.so.0.8.1 librz/bp/librz_bp.so.0.8.1 librz/syscall/librz_syscall.so.0.8.1 librz/egg/librz_egg.so.0.8.1 librz/search/librz_search.so.0.8.1 librz/sign/librz_sign.so.0.8.1 librz/il/librz_il.so.0.8.1 librz/arch/librz_arch.so.0.8.1 librz/type/librz_type.so.0.8.1 librz/debug/librz_debug.so.0.8.1 librz/config/librz_config.so.0.8.1 librz/bin/librz_bin.so.0.8.1 subprojects/rzgdb/librzgdb.a subprojects/rizin-shell-parser/libshell_parser.a '/packages/tree_sitter-0.20.10~git-1/.self/develop/lib/libtree-sitter.so' -lm -Wl,--end-group
FAILED: librz/core/librz_core.so.0.8.1 
cc  -o librz/core/librz_core.so.0.8.1 librz/core/librz_core.so.0.8.1.p/meson-generated_.._cmd_descs_cmd_descs.c.o librz/core/librz_core.so.0.8.1.p/agraph.c.o librz/core/librz_core.so.0.8.1.p/analysis_objc.c.o librz/core/librz_core.so.0.8.1.p/analysis_tp.c.o librz/core/librz_core.so.0.8.1.p/basefind.c.o librz/core/librz_core.so.0.8.1.p/cagraph.c.o librz/core/librz_core.so.0.8.1.p/cgraph.c.o librz/core/librz_core.so.0.8.1.p/canalysis.c.o librz/core/librz_core.so.0.8.1.p/cannotated_code.c.o librz/core/librz_core.so.0.8.1.p/carg.c.o librz/core/librz_core.so.0.8.1.p/casm.c.o librz/core/librz_core.so.0.8.1.p/cautocmpl.c.o librz/core/librz_core.so.0.8.1.p/cbin.c.o librz/core/librz_core.so.0.8.1.p/cbounds.c.o librz/core/librz_core.so.0.8.1.p/cconfig.c.o librz/core/librz_core.so.0.8.1.p/ccrypto.c.o librz/core/librz_core.so.0.8.1.p/cdebug.c.o librz/core/librz_core.so.0.8.1.p/cesil.c.o librz/core/librz_core.so.0.8.1.p/cfile.c.o librz/core/librz_core.so.0.8.1.p/cflag.c.o librz/core/librz_core.so.0.8.1.p/chash.c.o librz/core/librz_core.so.0.8.1.p/cheap.c.o librz/core/librz_core.so.0.8.1.p/cil.c.o librz/core/librz_core.so.0.8.1.p/cio.c.o librz/core/librz_core.so.0.8.1.p/clang.c.o librz/core/librz_core.so.0.8.1.p/cmeta.c.o librz/core/librz_core.so.0.8.1.p/cmp.c.o librz/core/librz_core.so.0.8.1.p/core.c.o librz/core/librz_core.so.0.8.1.p/cparser.c.o librz/core/librz_core.so.0.8.1.p/cpdb.c.o librz/core/librz_core.so.0.8.1.p/cplugin.c.o librz/core/librz_core.so.0.8.1.p/cprint.c.o librz/core/librz_core.so.0.8.1.p/creg.c.o librz/core/librz_core.so.0.8.1.p/csearch.c.o librz/core/librz_core.so.0.8.1.p/csign.c.o librz/core/librz_core.so.0.8.1.p/ctypes.c.o librz/core/librz_core.so.0.8.1.p/cvfile.c.o librz/core/librz_core.so.0.8.1.p/csyscall.c.o librz/core/librz_core.so.0.8.1.p/disasm.c.o librz/core/librz_core.so.0.8.1.p/fortune.c.o librz/core/librz_core.so.0.8.1.p/golang.c.o librz/core/librz_core.so.0.8.1.p/hack.c.o librz/core/librz_core.so.0.8.1.p/libs.c.o librz/core/librz_core.so.0.8.1.p/linux_heap_glibc.c.o librz/core/librz_core.so.0.8.1.p/linux_heap_glibc64.c.o librz/core/librz_core.so.0.8.1.p/rop.c.o librz/core/librz_core.so.0.8.1.p/project.c.o librz/core/librz_core.so.0.8.1.p/project_migrate.c.o librz/core/librz_core.so.0.8.1.p/rtr.c.o librz/core/librz_core.so.0.8.1.p/seek.c.o librz/core/librz_core.so.0.8.1.p/serialize_core.c.o librz/core/librz_core.so.0.8.1.p/task.c.o librz/core/librz_core.so.0.8.1.p/yank.c.o librz/core/librz_core.so.0.8.1.p/p_core_dex.c.o librz/core/librz_core.so.0.8.1.p/p_core_java.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_alias.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_api.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_block.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_cmp.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_egg.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_eval.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_flag.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_flirt.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_hash.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_history.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_info.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interactive.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interactive_panel.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_interpret.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_macro.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_meta.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_open.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_plugins.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_project.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_query.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_quit.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_remote.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_resize.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_seek.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_shell.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_system.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_tasks.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_type.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_write.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_yank.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_regs.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_linux_heap_glibc.c.o librz/core/librz_core.so.0.8.1.p/cmd_cmd_windows_heap.c.o librz/core/librz_core.so.0.8.1.p/tui_biteditor.c.o librz/core/librz_core.so.0.8.1.p/tui_classes.c.o librz/core/librz_core.so.0.8.1.p/tui_colors.c.o librz/core/librz_core.so.0.8.1.p/tui_comments.c.o librz/core/librz_core.so.0.8.1.p/tui_config.c.o librz/core/librz_core.so.0.8.1.p/tui_define.c.o librz/core/librz_core.so.0.8.1.p/tui_esil.c.o librz/core/librz_core.so.0.8.1.p/tui_flags.c.o librz/core/librz_core.so.0.8.1.p/tui_hud.c.o librz/core/librz_core.so.0.8.1.p/tui_modes.c.o librz/core/librz_core.so.0.8.1.p/tui_visual.c.o librz/core/librz_core.so.0.8.1.p/tui_vmenus.c.o librz/core/librz_core.so.0.8.1.p/tui_vmenus_graph.c.o librz/core/librz_core.so.0.8.1.p/tui_vasm.c.o librz/core/librz_core.so.0.8.1.p/tui_vmarks.c.o librz/core/librz_core.so.0.8.1.p/tui_panels.c.o librz/core/librz_core.so.0.8.1.p/tui_tabs.c.o librz/core/librz_core.so.0.8.1.p/tui_rop.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,-soname,librz_core.so.0.8 '-Wl,-rpath,$ORIGIN/../util:$ORIGIN/../demangler:$ORIGIN/../diff:$ORIGIN/../magic:$ORIGIN/../socket:$ORIGIN/../flag:$ORIGIN/../cons:$ORIGIN/../lang:$ORIGIN/../hash:$ORIGIN/../crypto:$ORIGIN/../io:$ORIGIN/../reg:$ORIGIN/../bp:$ORIGIN/../syscall:$ORIGIN/../egg:$ORIGIN/../arch:$ORIGIN/../search:$ORIGIN/../config:$ORIGIN/../bin:$ORIGIN/../type:$ORIGIN/../il:$ORIGIN/../sign:$ORIGIN/../debug:/packages/tree_sitter-0.20.10~git-1/.self/develop/lib:/packages/libpcre2-10.45-1/.self/develop/lib:/packages/zlib-1.3.1-4/.self/develop/lib:/packages/xz_utils-5.6.2-2/.self/develop/lib:/packages/openssl3-3.5.0-2/.self/develop/lib:/packages/file-5.43-2/.self/develop/lib:/packages/xxhash-0.8.3-1/.self/develop/lib:/packages/libzip-1.9.2-2/.self/develop/lib:/packages/capstone5-5.0.1-2/.self/develop/lib:/packages/zstd-1.5.6-2/.self/develop/lib:/packages/libmspack-0.10.1~alpha-2/.self/develop/lib' -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/util -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/demangler -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/diff -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/magic -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/socket -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/flag -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/cons -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/lang -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/hash -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/crypto -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/io -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/reg -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/bp -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/syscall -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/egg -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/arch -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/search -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/config -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/bin -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/type -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/il -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/sign -Wl,-rpath-link,/sources/rizin-0.8.1/build/librz/debug '-Wl,-rpath-link,/packages/tree_sitter-0.20.10~git-1/.self/develop/lib' -Wl,-rpath-link,/packages/libpcre2-10.45-1/.self/develop/lib -Wl,-rpath-link,/packages/zlib-1.3.1-4/.self/develop/lib -Wl,-rpath-link,/packages/xz_utils-5.6.2-2/.self/develop/lib -Wl,-rpath-link,/packages/openssl3-3.5.0-2/.self/develop/lib -Wl,-rpath-link,/packages/file-5.43-2/.self/develop/lib -Wl,-rpath-link,/packages/xxhash-0.8.3-1/.self/develop/lib -Wl,-rpath-link,/packages/libzip-1.9.2-2/.self/develop/lib -Wl,-rpath-link,/packages/capstone5-5.0.1-2/.self/develop/lib -Wl,-rpath-link,/packages/zstd-1.5.6-2/.self/develop/lib '-Wl,-rpath-link,/packages/libmspack-0.10.1~alpha-2/.self/develop/lib' -Wl,--start-group librz/util/librz_util.so.0.8.1 librz/demangler/librz_demangler.so.0.8.1 librz/diff/librz_diff.so.0.8.1 librz/magic/librz_magic.so.0.8.1 librz/socket/librz_socket.so.0.8.1 librz/flag/librz_flag.so.0.8.1 librz/cons/librz_cons.so.0.8.1 librz/lang/librz_lang.so.0.8.1 librz/hash/librz_hash.so.0.8.1 librz/crypto/librz_crypto.so.0.8.1 librz/io/librz_io.so.0.8.1 librz/reg/librz_reg.so.0.8.1 librz/bp/librz_bp.so.0.8.1 librz/syscall/librz_syscall.so.0.8.1 librz/egg/librz_egg.so.0.8.1 librz/search/librz_search.so.0.8.1 librz/sign/librz_sign.so.0.8.1 librz/il/librz_il.so.0.8.1 librz/arch/librz_arch.so.0.8.1 librz/type/librz_type.so.0.8.1 librz/debug/librz_debug.so.0.8.1 librz/config/librz_config.so.0.8.1 librz/bin/librz_bin.so.0.8.1 subprojects/rzgdb/librzgdb.a subprojects/rizin-shell-parser/libshell_parser.a '/packages/tree_sitter-0.20.10~git-1/.self/develop/lib/libtree-sitter.so' -lm -Wl,--end-group
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/13.3.0/../../../../x86_64-unknown-haiku/bin/ld: librz/core/librz_core.so.0.8.1.p/linux_heap_glibc.c.o: in function `resolve_heap_tcache_32':
linux_heap_glibc.c:(.text+0x609f): undefined reference to `rz_debug_get_tls'
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/13.3.0/../../../../x86_64-unknown-haiku/bin/ld: linux_heap_glibc.c:(.text+0x60e8): undefined reference to `rz_debug_native_threads'
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/13.3.0/../../../../x86_64-unknown-haiku/bin/ld: librz/core/librz_core.so.0.8.1.p/linux_heap_glibc64.c.o: in function `resolve_heap_tcache_64':
linux_heap_glibc64.c:(.text+0x60f0): undefined reference to `rz_debug_get_tls'
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/13.3.0/../../../../x86_64-unknown-haiku/bin/ld: linux_heap_glibc64.c:(.text+0x6138): undefined reference to `rz_debug_native_threads'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1.
cleaning 'chroot/boot' folder
keeping chroot folder /Opslag/haikuports/dev-util/rizin/work-0.8.1 intact for inspection
Error: Build has failed - stopping.
```

--- a/librz/arch/isa/tricore/tricore_il.c
+++ b/librz/arch/isa/tricore/tricore_il.c
@@ -2588,10 +2588,10 @@ static RzILOpEffect *e_BS(RzAsmTriCoreContext *ctx, unsigned b, FUNC_BS f) {
 	return NULL;
 }
 
-#define MIN(x, y)        ITE(UGT(x, y), DUP(x), DUP(y))
+#define IL_MIN(x, y)        ITE(UGT(x, y), DUP(x), DUP(y))
 #define CRC_32_GENERATOR 0xEDB88320
 static RzILOpPure *crc_div(RzILOpPure *c, RzILOpPure *g, RzILOpPure *crc_width, RzILOpPure *data_width) {
-	return LET("shift", MIN(crc_width, data_width),
+	return LET("shift", IL_MIN(crc_width, data_width),
 		MOD(SHIFTL0(c, VARLP("shift")), LOGOR(g, SHIFTL0(U32(1), DUP(crc_width)))));
 }
 

--- a/librz/arch/isa/tricore/tricore_il.c
+++ b/librz/arch/isa/tricore/tricore_il.c
@@ -2588,7 +2588,7 @@ static RzILOpEffect *e_BS(RzAsmTriCoreContext *ctx, unsigned b, FUNC_BS f) {
 	return NULL;
 }
 
-#define IL_MIN(x, y)        ITE(UGT(x, y), DUP(x), DUP(y))
+#define IL_MIN(x, y)     ITE(UGT(x, y), DUP(x), DUP(y))
 #define CRC_32_GENERATOR 0xEDB88320
 static RzILOpPure *crc_div(RzILOpPure *c, RzILOpPure *g, RzILOpPure *crc_width, RzILOpPure *data_width) {
 	return LET("shift", IL_MIN(crc_width, data_width),

--- a/librz/cons/input.c
+++ b/librz/cons/input.c
@@ -103,6 +103,7 @@ RZ_API int rz_cons_arrow_to_hjkl(int ch) {
 			ch = 0xf1 + (ch & 0xf);
 			break;
 		}
+		// fall-thru
 	case '[': // 0x5b function keys (2)
 		/* Haiku need ESC + [ for PageUp and PageDown  */
 		if (ch < 'A' || ch == '[') {

--- a/librz/debug/p/debug_native.c
+++ b/librz/debug/p/debug_native.c
@@ -30,6 +30,7 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #warning "Unsupported debugging platform"
 #undef DEBUGGER
 #define DEBUGGER 0
+#include "native/unsupported.c"
 #endif
 
 #elif __ANDROID__
@@ -43,6 +44,7 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #warning "Unsupported debugging platform"
 #undef DEBUGGER
 #define DEBUGGER 0
+#include "native/unsupported.c"
 #endif
 
 #elif __linux__
@@ -74,6 +76,7 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #warning "Unsupported debugging platform"
 #undef DEBUGGER
 #define DEBUGGER 0
+#include "native/unsupported.c"
 #endif
 
 #elif __sun
@@ -81,11 +84,12 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #undef DEBUGGER
 #define DEBUGGER 0
 #warning "No debugger support for SunOS yet"
-
+#include "native/unsupported.c"
 #else
 #warning "Unsupported debugging platform"
 #undef DEBUGGER
 #define DEBUGGER 0
+#include "native/unsupported.c"
 #endif // Native OS & Arch
 
 #if DEBUGGER

--- a/librz/debug/p/native/unsupported.c
+++ b/librz/debug/p/native/unsupported.c
@@ -1,0 +1,16 @@
+// SPDX-FileCopyrightText: 2025 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2025 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+/**
+ * \file This is a file with dummy debugger functions for when
+ *       the native debugger is unsupported.
+ */
+
+RZ_API ut64 rz_debug_get_tls(RZ_NONNULL RzDebug *dbg, int tid) {
+	return 0;
+}
+
+RZ_API RZ_OWN RzList /*<RzDebugPid *>*/ *rz_debug_native_threads(RzDebug *dbg, int pid) {
+	return rz_list_new();
+}

--- a/librz/debug/p/native/unsupported.c
+++ b/librz/debug/p/native/unsupported.c
@@ -14,3 +14,17 @@ RZ_API ut64 rz_debug_get_tls(RZ_NONNULL RzDebug *dbg, int tid) {
 RZ_API RZ_OWN RzList /*<RzDebugPid *>*/ *rz_debug_native_threads(RzDebug *dbg, int pid) {
 	return rz_list_new();
 }
+
+static int rz_debug_native_continue(RzDebug *dbg, int pid, int tid, int sig) {
+	return -1;
+}
+
+static int rz_debug_native_reg_read(RzDebug *dbg, int type, ut8 *buf, int size) {
+	// TODO: this is returning a boolean, but for some reasons we have int
+	return false;
+}
+
+static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int size) {
+	// TODO: this is returning a boolean, but for some reasons we have int
+	return false;
+}

--- a/librz/include/rz_debug.h
+++ b/librz/include/rz_debug.h
@@ -472,10 +472,6 @@ RZ_API int rz_debug_continue_pass_exception(RzDebug *dbg);
 
 /* process/thread handling */
 RZ_API bool rz_debug_select(RzDebug *dbg, int pid, int tid);
-// RZ_API int rz_debug_pid_add(RzDebug *dbg);
-// RZ_API int rz_debug_pid_add_thread(RzDebug *dbg);
-// RZ_API int rz_debug_pid_del(RzDebug *dbg);
-// RZ_API int rz_debug_pid_del_thread(RzDebug *dbg);
 RZ_API RzDebugPid *rz_debug_pid_new(const char *path, int pid, int uid, char status, ut64 pc);
 RZ_API RzDebugPid *rz_debug_pid_free(RzDebugPid *pid);
 RZ_API RzList /*<RzDebugPid *>*/ *rz_debug_pids(RzDebug *dbg, int pid);
@@ -502,8 +498,6 @@ RZ_API int rz_debug_signal_set(RzDebug *dbg, int num, ut64 addr);
 RZ_API bool rz_debug_can_kill(RzDebug *dbg);
 RZ_API int rz_debug_kill(RzDebug *dbg, int pid, int tid, int sig);
 RZ_API RzList /*<void *>*/ *rz_debug_kill_list(RzDebug *dbg);
-// XXX: must be uint64 action
-RZ_API int rz_debug_kill_setup(RzDebug *dbg, int sig, int action);
 
 /* handle.c */
 RZ_API void rz_debug_plugin_init(RzDebug *dbg);
@@ -531,7 +525,6 @@ RZ_API int rz_debug_desc_dup(RzDebug *dbg, int fd, int newfd);
 RZ_API int rz_debug_desc_read(RzDebug *dbg, int fd, ut64 addr, int len);
 RZ_API int rz_debug_desc_seek(RzDebug *dbg, int fd, ut64 addr); // TODO: whence?
 RZ_API int rz_debug_desc_write(RzDebug *dbg, int fd, ut64 addr, int len);
-RZ_API int rz_debug_desc_list(RzDebug *dbg, int rad);
 
 /* registers */
 RZ_API bool rz_debug_reg_profile_sync(RzDebug *dbg);
@@ -588,7 +581,6 @@ RZ_API int rz_debug_esil_watch_empty(RzDebug *dbg);
 RZ_API void rz_debug_esil_prestep(RzDebug *d, int p);
 
 /* record & replay */
-// RZ_API ut8 rz_debug_get_byte(RzDebug *dbg, ut32 cnum, ut64 addr);
 RZ_API bool rz_debug_add_checkpoint(RzDebug *dbg);
 RZ_API bool rz_debug_session_add_reg_change(RzDebugSession *session, int arena, ut64 offset, ut64 data);
 RZ_API bool rz_debug_session_add_mem_change(RzDebugSession *session, ut64 addr, ut8 data);
@@ -635,25 +627,5 @@ static inline void *rz_debug_ptrace_func(RzDebug *dbg, void *(*func)(void *), vo
 #ifdef __cplusplus
 }
 #endif
-
-#endif
-
-/* regset */
-// RZ_API struct rz_regset_t* rz_regset_diff(struct rz_regset_t *a, struct rz_regset_t *b);
-// RZ_API int rz_regset_set(struct rz_regset_t *r, int idx, const char *name, ut64 value);
-// RZ_API struct rz_regset_t *rz_regset_new(int size);
-// RZ_API void rz_regset_free(struct rz_regset_t *r);
-
-#if 0
-Missing callbacks
-=================
- - alloc
- - dealloc
- - list maps (memory regions)
- - change memory protections
- - touchtrace
- - filedescriptor set/get/mod..
- - get/set signals
- - get regs, set regs
 
 #endif

--- a/librz/util/sys.c
+++ b/librz/util/sys.c
@@ -1229,12 +1229,11 @@ RZ_API char *rz_sys_pid_to_path(int pid) {
 	}
 	return rz_str_dup(pathbuf);
 #else
-	int ret;
 #if __FreeBSD__ || __DragonFly__
 	char pathbuf[PATH_MAX];
 	size_t pathbufl = sizeof(pathbuf);
 	int mib[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, pid };
-	ret = sysctl(mib, 4, pathbuf, &pathbufl, NULL, 0);
+	int ret = sysctl(mib, 4, pathbuf, &pathbufl, NULL, 0);
 	if (ret != 0) {
 		return NULL;
 	}
@@ -1245,7 +1244,7 @@ RZ_API char *rz_sys_pid_to_path(int pid) {
 	size_t len;
 
 	pathbuf[0] = '\0';
-	ret = sysctl(mib, 4, NULL, &len, NULL, 0);
+	int ret = sysctl(mib, 4, NULL, &len, NULL, 0);
 	if (ret < 0) {
 		return NULL;
 	}
@@ -1310,7 +1309,7 @@ RZ_API char *rz_sys_pid_to_path(int pid) {
 #else
 	char buf[128], pathbuf[1024];
 	snprintf(buf, sizeof(buf), "/proc/%d/exe", pid);
-	ret = readlink(buf, pathbuf, sizeof(pathbuf) - 1);
+	int ret = readlink(buf, pathbuf, sizeof(pathbuf) - 1);
 	if (ret < 1) {
 		return NULL;
 	}

